core/src/syscall/precompiles/blake2b/compress/mod.rs

```diff
@@ -0,0 +1,198 @@
+//! This module contains the implementation of the `blake2b inner compress` precompile based on
+//! the implementation of the `blake2b` hash function in BLAKE2.
+//!
+//! Pseudo-code.
+//!
+//! state = [0u64; 16]
+//! message_chunk = [0u64; 16]
+//!
+//! for round in 0..12 {
+//!     for operation in 0..8 {
+//!         // * Pick 4 indices a, b, c, d for the state, based on the operation index.
+//!         // * Pick 2 indices x, y for the message, based on both the round and the operation index.
+//!         //
+//!         // g takes those 6 values, and updates the 4 state values, at indices a, b, c, d.
+//!         //
+//!         // Each call of mix becomes one row in the trace.
+//!         mix(&mut state[a], &mut state[b], &mut state[c], &mut state[d], message[x], message[y]);
+//!     }
+//! }
+use crate::cpu::{MemoryReadRecord, MemoryWriteRecord};
+mod air;
+mod columns;
+mod execute;
+mod mix;
+mod trace;
+
+/// The number of `u64`s in the message of the compress inner operation.
+pub(crate) const MSG_SIZE: usize = 16;
+
+/// Each msg word is 8 bytes and our words size is 4 bytes. So we need to double the size of the
+/// message.
+#[allow(dead_code)]
+pub(crate) const MSG_NUM_WORDS: usize = MSG_SIZE * 2;
+
+/// The number of rounds in the compress inner operation.
+pub(crate) const NUM_MIX_ROUNDS: usize = 12;
+
+/// The number of time we call `mix` in the compress inner operation in each mix round.
+pub(crate) const OPERATION_COUNT: usize = 8;
+
+/// The number of `Word`s in the state that we pass to `mix`. Each `Word` is 8 bytes.
```

Comment:
nit: is "Word" here correct? don't you mean u64?

Reply:
maybe it's better to just call this `STATE_SIZE` to be consistent with `MSG_SIZE`